leasing/report/lease/reservations.py

```diff
@@ -63,6 +63,87 @@ def get_reservation_procedure(obj):
     return obj.reservation_procedure.name if obj.reservation_procedure else None
 
 
+def get_plan_unit_intended_uses(obj):
+    plan_units_intended_uses = []
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+        for plan_unit in lease_area.plan_units.all():
+            if plan_unit.plan_unit_intended_use:
+                plan_units_intended_uses.append(plan_unit.plan_unit_intended_use.name)
+
+    return ", ".join(plan_units_intended_uses)
+
+
+def get_plan_unit_detailed_plan_identifier(obj):
+    plan_units_detailed_plan_identifiers = []
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+        for plan_unit in lease_area.plan_units.all():
+            if plan_unit.detailed_plan_identifier:
+                plan_units_detailed_plan_identifiers.append(
+                    plan_unit.detailed_plan_identifier
+                )
+
+    return ", ".join(plan_units_detailed_plan_identifiers)
+
+
+def get_plan_unit_usage_distribution_main_build_permission(obj):
+    """
+    build permission of usage distribution with highest build permission
+    """
+    main_build_permission = 0
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+        for plan_unit in lease_area.plan_units.all():
+            for usage_distribution in plan_unit.usage_distributions.all():
+                if int(usage_distribution.build_permission) > main_build_permission:
+                    main_build_permission = int(usage_distribution.build_permission)
+
+    if main_build_permission != 0:
+        return str(main_build_permission)
+    else:
+        return ""
+
+
+def get_plan_unit_usage_distribution_other_build_permission(obj):
+    """
+    aggregate build permission of all usage distributions except usage distribution with highest build permission
+    """
+    other_build_permission = 0
+    highest_build_permission = 0
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+        for plan_unit in lease_area.plan_units.all():
+            for usage_distribution in plan_unit.usage_distributions.all():
+                if usage_distribution.build_permission.isdigit():
```

Comment:
is build_permission always a `str`?

Reply:
In the database it is string type, though I think it would make more sense as numeric type. I found in the data that sometimes build_permission is equal to "-" but otherwise number.

Reply:
Can it be `NULL`?, so can the field be `nullable=True`

Reply:
At the moment its not nullable